core/src/health.rs

```diff
@@ -0,0 +1,214 @@
+use std::{
+    net::SocketAddr,
+    sync::Arc,
+};
+
+use axum::{
+    extract::State,
+    http::StatusCode,
+    response::Json,
+    routing::get,
+    Router,
+};
+use serde::{Deserialize, Serialize};
+use tokio::net::TcpListener;
+use tracing::{error, info};
+
+use crate::{
+    database::postgres::client::PostgresClient,
+    indexer::task_tracker::active_indexing_count,
+    manifest::core::Manifest,
+    system_state::is_running,
+};
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct HealthStatus {
+    pub status: String,
+    pub timestamp: String,
+    pub services: HealthServices,
+    pub indexing: IndexingStatus,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct HealthServices {
+    pub database: String,
+    pub indexing: String,
+    pub sync: String,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct IndexingStatus {
+    pub active_tasks: usize,
+    pub is_running: bool,
+}
+
+#[derive(Clone)]
+pub struct HealthServerState {
+    pub manifest: Arc<Manifest>,
+    pub postgres_client: Option<Arc<PostgresClient>>,
+}
+
+pub struct HealthServer {
+    port: u16,
+    state: HealthServerState,
+}
+
+impl HealthServer {
+    pub fn new(port: u16, manifest: Arc<Manifest>, postgres_client: Option<Arc<PostgresClient>>) -> Self {
+        Self {
+            port,
+            state: HealthServerState {
+                manifest,
+                postgres_client,
+            },
+        }
+    }
+
+    pub async fn start(self) -> Result<(), Box<dyn std::error::Error + Send + Sync>> {
+        let app = Router::new()
+            .route("/health", get(health_handler))
+            .with_state(self.state);
+
+        let addr = SocketAddr::from(([0, 0, 0, 0], self.port));
+        let listener = TcpListener::bind(addr).await?;
+
+        info!("🩺 Health server started on http://0.0.0.0:{}/health", self.port);
+
+        axum::serve(listener, app).await?;
+        Ok(())
+    }
+}
+
+async fn health_handler(State(state): State<HealthServerState>) -> Result<Json<HealthStatus>, StatusCode> {
```

Comment:
this function should split up and then build the object with the function calls its to long hard to read and way to many if elses

```
HealthStatus {
        status: "healthy".to_string(),
        timestamp: chrono::Utc::now().to_rfc3339(),
        services: HealthServices {
            database: database_health_check(),
            ...
        },
        indexing: IndexingStatus {
            active_tasks: active_indexing_count(),
            is_running: is_running(),
        },
    }
   ```